target_chains/stylus/contracts/pyth-receiver/src/lib.rs

```diff
@@ -131,9 +353,30 @@ impl PythReceiver {
     }
 
     fn is_no_older_than(&self, publish_time: U64, max_age: u64) -> bool {
-        let current_u64: u64 = self.vm().block_timestamp();
-        let publish_time_u64: u64 = publish_time.to::<u64>();
-        
-        current_u64.saturating_sub(publish_time_u64) <= max_age
+        self.get_current_timestamp()
+            .saturating_sub(publish_time.to::<u64>())
+            <= max_age
+    }
+
+    // Stylus doesn't provide a way to mock up the testing timestamp
+    // so at the moment I'm using the testing trait to let me test old timestamps
+    fn get_current_timestamp(&self) -> u64 {
+        #[cfg(test)]
+        {
+            1761573860u64
```

Comment:
you can use libraries like `mock_instant` to make this more configurable for various test cases.